app/upgrades/v17/upgrades.go

```diff
@@ -101,10 +103,36 @@ func CreateUpgradeHandler(
 			return vm, errorsmod.Wrapf(err, "unable to add rate limits to Osmosis")
 		}
 
+		ctx.Logger().Info("Disabling tokenization on the hub...")
+		DisableTokenization(ctx, stakeibcKeeper, "cosmoshub-4")
+
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
 
+// Sends the ICA message which disables LSM style tokenization of shares from the delegation
+// account for this chain as a security to prevent possibility of large/fast withdrawls
+func DisableTokenization(ctx sdk.Context, k stakeibckeeper.Keeper, chainId string) error {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		if hostZone.ChainId == chainId {
+			// Build the msg for the disable tokenization ICA tx
+			var msgs []proto.Message
+			msgs = append(msgs, &stakeibctypes.MsgDisableTokenizeShares{
+				DelegatorAddress: hostZone.DelegationIcaAddress,
+			})
+				
+			// Send the ICA tx to disable tokenization
+			timeoutTimestamp := uint64(time.Now().Add(24 * time.Hour).Nanosecond())
```

Comment:
```suggestion
			timeoutTimestamp := uint64(ctx.BlockTime().Add(24 * time.Hour).Nanosecond())
```

Reply:
So we do have access to the block time and the epoch info when running the upgrade handler, it is just unwise to depend on the next epoch as the timeout point because this could be run at any point through the epoch and could be close to the end. Makes sense